python/ray/dag/dag_node.py

```diff
@@ -199,6 +199,12 @@ def experimental_compile(
         if _max_buffered_results is None:
             _max_buffered_results = ctx.max_buffered_results
 
+        # Validate whether this DAG node has already been compiled.
+        if self.is_output_node:
+            raise ValueError(
+                "The DAG has already been compiled! "
+                "Please reuse the existing compiled DAG."
```

Comment:
Can you add a comment "it is not allowed to call .teardown() and call .compile again on the same dag object. Reuse the existing compiled DAG or create a new one"

Reply:
Update: https://github.com/ray-project/ray/pull/47431/commits/c31286deac9f2a6d8ff996e049675685d76d479f